wd_helper.au3

```diff
@@ -796,24 +796,22 @@ Func _WD_Screenshot($sSession, $sElement = Default, $iOutputType = Default)
 	$iErr = @error
 
 	If $iErr = $_WD_ERROR_Success  Then
-		If $iOutputType < 3 Then
-			$dBinary = __WD_Base64Decode($sResponse)
-			If @error Then $iErr = $_WD_ERROR_GeneralError
-		EndIf
-
-		If $iErr = $_WD_ERROR_Success Then ; Recheck after __WD_Base64Decode() usage
+		If $iOutputType < 3 Then $dBinary = __WD_Base64Decode($sResponse)
+		If @error Then ; Recheck after __WD_Base64Decode() usage
+			$iErr = $_WD_ERROR_GeneralError
+		Else
```

Comment:
Was there something wrong with the prior code structure? How is this "better" or more efficient?

Reply:
Shorter. There was no need to again wirte/check this long line:
`If $iErr = $_WD_ERROR_Success Then`

Reply:
Ok, but shorter doesn't always mean better. There needs to be a balance between simplicity, readability, and maintainability. 

FWIW, I sometimes have trouble following / understanding this type of coding. The value of @error could be coming from somewhere other than __WD_Base64Decode, so this gets "fuzzy" to me. I know that it should be 0 because of the prior `If $iErr = $_WD_ERROR_Success  Then`, but it just doesn't "feel" right to me.

For example, what happens if I add a function call prior to `If $iOutputType < 3 Then $dBinary = __WD_Base64Decode($sResponse)`? This function could return an error, so @error wouldn't be zero and the existing logic would fail.

With the prior structure, I don't believe this same failure would occur.

Reply:
refactoring reverted